bftengine/include/bftengine/Replica.hpp

```diff
@@ -32,7 +32,8 @@ enum MsgFlag : uint8_t {
   EMPTY_FLAGS = 0x0,
   READ_ONLY_FLAG = 0x1,
   PRE_PROCESS_FLAG = 0x2,
-  HAS_PRE_PROCESSED_FLAG = 0x4
+  HAS_PRE_PROCESSED_FLAG = 0x4,
+  KEY_EXCHANGE_FLAG = 0x8
```

Comment:
Why not creating a flag for all internal bft commands something like BFT_INTENAL_CMD_FLAG=0X8

Reply:
Then we need another mechanism to differentiate between msgs, it can be done, but not necessary at this phase IMO